### PR TITLE
Feature 2294/notify managers that there are review assignments awaiting review

### PR DESCRIPTION
**Pull request by montesmoci**

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java [hunk 1]
```diff
@@ -59,6 +60,14 @@ List<MemberProfile> search(@Nullable String firstName, @Nullable String middleNa
...
 
     List<MemberProfile> findAll();
+    List<UUID> findSupervisoridByIdIn(Set<UUID> Ids);
+
+    @Query(value = """
+            SELECT PGP_SYM_DECRYPT(cast(workEmail as bytea), :aesKey) as workEmail
+                    FROM member_profile mp
+                    WHERE  mp.id IN (:ids)""",
+            nativeQuery = true)
+    List<String> findWorkEmailByIdIn(String aesKey, Set<String> ids);
```
Comment (timyates) [inline server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java:70]: Can we drop the aesKey parameter from here, and switch to use property expansion as with the other queries

```suggestion
    @Query(value = """
            SELECT PGP_SYM_DECRYPT(cast(workEmail as bytea), '${aes.key}') as workEmail
                    FROM member_profile mp
                    WHERE  mp.id IN (:ids)""",
            nativeQuery = true)
    List<String> findWorkEmailByIdIn(Set<String> ids);
```

And then we don't need to inject the value in to `ReviewPeriodServicesImpl` on line 41, nor pass it as a parameter on line 145

Comment (ZacharyKlein) [reply server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java:70]: Unfortunately, there appears to be a bug in Micronaut Data when using the property expansion syntax with `where in` queries. The injected property gets included with the method arguments as a SQL argument, throwing off the number of arguments.

Comment (timyates) [reply server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java:70]: Ahhhh! I see 👍

File: server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java [hunk 2]
```diff
@@ -21,14 +29,34 @@ class ReviewPeriodServicesImpl implements ReviewPeriodServices {
...
+        this.reviewAssignmentRepository = reviewAssignmentRepository;
+        this.memberProfileRepository = memberProfileRepository;
         this.reviewStatusTransitionValidator = reviewStatusTransitionValidator;
+        this.emailSender = emailSender;
+        this.environment = environment;
+        this.webAddress = webAddress;
+    }
+
+   void setEmailSender(EmailSender emailSender) {
+        this.emailSender = emailSender;
```
[the remark this answers is not in the document]
Comment (timyates) [reply server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java:59]: If agreed, we should then remove the `import io.micronaut.context.annotation.Value;` from the imports as it's unused 👍

[the remark this answers is not in the document]
Comment (timyates) [reply server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java:59]: I *think* I found a way to work around the bug, *and* avoid ghaving to pass the key as a parameter by using `@ParameterExpression` on the Repository method to grab the key from the environment 🤔 

```diff
diff --git a/server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java b/server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java
index 807acb0b5..cef6a4963 100644
--- a/server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java
+++ b/server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileRepository.java
@@ -1,6 +1,7 @@
 package com.objectcomputing.checkins.services.memberprofile;
 
 import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.annotation.ParameterExpression;
 import io.micronaut.data.annotation.Query;
 import io.micronaut.data.jdbc.annotation.JdbcRepository;
 import io.micronaut.data.model.query.builder.sql.Dialect;
@@ -67,7 +68,8 @@ public interface MemberProfileRepository extends CrudRepository<MemberProfile, U
                     FROM member_profile mp
                     WHERE  mp.id IN (:ids)""",
             nativeQuery = true)
-    List<String> findWorkEmailByIdIn(String aesKey, Set<String> ids);
+    @ParameterExpression(name = "aesKey", expression = "#{ env['aes.key'] }")
+    List<String> findWorkEmailByIdIn(Set<String> ids);
 
     @Query(value = "WITH RECURSIVE subordinate AS (SELECT " +
     "id, firstname, middlename, lastname, suffix, title, pdlid, location, workemail, employeeid, startdate, biotext, supervisorid, terminationdate, birthdate, voluntary, excluded, 0 as level " +
diff --git a/server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java b/server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java
index d64428f34..dcfdb8b3b 100644
--- a/server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java
+++ b/server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java
@@ -7,7 +7,6 @@ import com.objectcomputing.checkins.notifications.email.EmailSender;
 import com.objectcomputing.checkins.notifications.email.MailJetConfig;
 import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
 import io.micronaut.context.annotation.Property;
-import io.micronaut.context.annotation.Value;
 import io.micronaut.context.env.Environment;
 import jakarta.inject.Named;
 import jakarta.inject.Singleton;
@@ -37,15 +36,13 @@ class ReviewPeriodServicesImpl implements ReviewPeriodServices {
     private final String webAddress;
     public static final String WEB_ADDRESS = "check-ins.web-address";
 
-    @Value("${aes.key}")
-    private String aesKey;
-
     public ReviewPeriodServicesImpl(ReviewPeriodRepository reviewPeriodRepository,
                                     ReviewAssignmentRepository reviewAssignmentRepository,
                                     MemberProfileRepository memberProfileRepository,
                                     ReviewStatusTransitionValidator reviewStatusTransitionValidator,
                                     @Named(MailJetConfig.HTML_FORMAT) EmailSender emailSender,
-                                    Environment environment, @Property(name = WEB_ADDRESS) String webAddress) {
+                                    Environment environment,
+                                    @Property(name = WEB_ADDRESS) String webAddress) {
         this.reviewPeriodRepository = reviewPeriodRepository;
         this.reviewAssignmentRepository = reviewAssignmentRepository;
         this.memberProfileRepository = memberProfileRepository;
@@ -142,7 +139,7 @@ class ReviewPeriodServicesImpl implements ReviewPeriodServices {
         }
 
         Set<String> supervisorIdsToString = supervisorIds.stream().map(UUID::toString).collect(Collectors.toSet());
-        List<String> supervisorEmails = memberProfileRepository.findWorkEmailByIdIn(aesKey, supervisorIdsToString);
+        List<String> supervisorEmails = memberProfileRepository.findWorkEmailByIdIn(supervisorIdsToString);
 
         // send notification to supervisors
         String emailContent = constructEmailContent(reviewPeriodId, reviewPeriodName);
```

[the remark this answers is not in the document]
Comment (montesmoci) [reply server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java]: This worked, thank you!

**Review by timyates — APPROVED** [review]

**Review by mjperry91 — APPROVED** [review]

**Pull request merged**